admin/controllers/school.js

```diff
@@ -444,14 +444,49 @@ const getSelectPupilNotTakingCheck = async (req, res, next) => {
 
   return res.render('school/select-pupils-not-taking-check', {
     breadcrumbs: req.breadcrumbs(),
+    sortField,
+    sortDirection,
     attendanceCodes,
-    formData,
     pupilsList,
     htmlSortDirection,
     arrowSortDirection
   })
 }
 
+const savePupilNotTakingCheck = async (req, res, next) => {
+  res.locals.pageTitle = 'Save pupils not taking the check'
+  req.breadcrumbs(res.locals.pageTitle)
+
+  if (req.body.attendanceCode === undefined || req.body.pupil === undefined) {
+    return res.redirect('/school/pupils-not-taking-check/select-pupils')
+  }
+
+  const todayDate = moment(moment.now()).format()
+  let pupils = req.body.pupil
+  let pupilsProcessed
+
+  pupils.forEach(async (id) => {
+    let pupil = fetchOnePupil(id, req.user.School)
+    if (pupil) {
+      pupil.attendanceCode = {
+        _id: req.body.attendanceCode,
+        dateRecorded: new Date(todayDate),
+        byUser: req.user.UserName
```

Comment:
UserName is not unique - `EmailAddress` probably is however.  We don't have a pattern for this. It may be best to use both Name and EmailAddress fields as `byUserName` `byUserEmail`.

@GuyHarwood @GeorgeChatzigiannis - thoughts?

Reply:
use both.  we need to explore this in more detail.  have added a card to architecture backlog

Reply:
This is in the next PR (story 15595)